TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/util/RefSafeStreamUtils.java

```diff
@@ -0,0 +1,31 @@
+package org.opensearch.migrations.replay.util;
+
+import com.google.errorprone.annotations.MustBeClosed;
+import io.netty.util.ReferenceCounted;
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+public final class RefSafeStreamUtils {
+    @MustBeClosed
+    public static <T, R extends ReferenceCounted> Stream<R> refSafeMap(Stream<T> inputStream,
+        Function<T, R> referenceTrackedMappingFunction) {
+        final Deque<R> refCountedTracker = new ArrayDeque<>();
+        return inputStream.map(t -> {
+            var resource = referenceTrackedMappingFunction.apply(t);
+            refCountedTracker.add(resource);
+            return resource;
+        }).onClose(() -> refCountedTracker.forEach(ReferenceCounted::release));
```

Comment:
For documentation reasons, I'd love to see a unit test that throws an exception during the stream and all of the refCounted objects properly deleted

Reply:
Added RefSafeStreamUtilsTest with comprehensive testing